packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -245,6 +246,37 @@ export class Pipeline extends cdk.Construct implements events.IEventRuleTarget {
     this.stages.splice(index, 0, stage);
   }
 
+  // ignore unused private method (it's actually used in Stage)
+  // @ts-ignore
+  private _generateOutputArtifactName(stage: actions.IStage, action: actions.Action): string {
+    // for now, just return a generic output artifact,
+    // ignoring the names of both the Stage, and the Action
+    return 'Artifact_' + (++this.artifactsCounter);
```

Comment:
The drawback is that this makes artifact names dependent on generation order. This means a user refactoring code & trying to ensure they didn't inadvertently change their infrastructure cannot rely on a simple diff test.

I would like this to be order-independent unless we have a very good reason why it's better otherwise (I understand this is a simplicity vs. user-friendliness tradeoff).

Reply:
I see. What would you see as the artifact name then? Some combination of Stage & Action names? `` `Artifact_${stage.name}_${action.name}` ``?

Reply:
We can use the `uniqueId `of the construct.

Reply:
Changed.

Reply:
One disadvantage of this: the artifact names are now super ugly (`Artifact_awscdkcodepipelinecodecommitcodebuildMyBuildProjectbuild61B48DC4`).

Reply:
But does anyone care?